server/models/bookingMethods/bookingBeforeSaveOperationHook.js

```diff
@@ -0,0 +1,139 @@
+const moment = require("moment-timezone");
+const { calendarService } = require("../../services/calendarService");
+const { log } = require("../../libraries/log");
+
+const getActualDate = () =>
+  moment()
+    .utc()
+    .format();
+
+// TODO: have a some sort of config file for setting custom office preferences
+const officeHours = {
+  timezone: "America/Mexico_City",
+  // Ideally we would write this like, { from: "monday": to: "friday" };
+  workingDays: [1, 2, 3, 4, 5],
+  workingHours: [
+    {
+      from: "08:00",
+      to: "18:00"
+    }
+  ]
+};
+
+const onlyInOfficeHours = (start, end, timezone = "America/Mexico_City") => {
+  const startDate = moment(start).tz(timezone);
+  const endDate = moment(end).tz(timezone);
+
+  // Check if is weekday
+  const isAvailableDay = day => officeHours.workingDays.includes(day);
```

Comment:
move functions outside `onlyInOfficeHours`

Reply:
I convert them to `pure functions`